addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/FormHandlerParam.java

```diff
@@ -211,7 +224,10 @@ public List<String> getEnabledFieldsNames() {
      */
     public String getEnabledFieldValue(String name) {
         for (FormHandlerParamField field : fields) {
-            if (field.getName().equalsIgnoreCase(name) && field.isEnabled()) {
+            if (!field.isRegex() && field.getName().equalsIgnoreCase(name) && field.isEnabled()) {
```

Comment:
Should I split these out into sub-methods so that the two different kinds can be checked in a more predictable order in the case of a simple match being the same as or a sub-set of a regex?

Reply:
Note this should be across all 3 or 4 lines here but I couldn’t comment multiple lines on my phone.